torchgeo/datasets/cloud_cover.py

```diff
@@ -0,0 +1,410 @@
+                source_item_hrefs = [
+                    os.path.join(self.root, link["href"].replace("../../", ""))
+                    for link in label_data["links"]
+                    if link["rel"] == "source"
+                ]
+                source_item_hrefs = sorted(source_item_hrefs)
+                source_item_paths = []
+
+                for item_href in source_item_hrefs:
+                    source_item_path = os.path.split(item_href)[0]
+                    if self._validate_stac_object(item_href):
+                        source_data = self._read_json_data(item_href)
+                        source_item_assets = [
+                            os.path.join(source_item_path, asset_value["href"])
+                            for asset_key, asset_value in source_data["assets"].items()
+                            if asset_key in self.bands
+                        ]
+                        source_item_assets = sorted(source_item_assets)
+                        for source_item_asset in source_item_assets:
+                            if self._validate_stac_object(source_item_asset):
+                                source_item_paths.append(source_item_asset)
+
+            item_meta["source"] = source_item_paths
+        return item_meta
+
+    def _load_collections(self) -> List[Dict[str, Any]]:
+        """Loads the paths to source and label assets for each collection.
+
+        Returns:
+            a dictionary with lists of filepaths to all assets for each chip/item
+
+        Raises:
+            RuntimeError if collection.json is not found in the uncompressed dataset
+        """
+        indexed_chips = []
+        label_collection = [
+            c for c in self.collection_names[self.split] if c.__contains__("label")
+        ][0]
+        label_collection_path = os.path.join(self.root, label_collection)
+        label_collection_json = os.path.join(label_collection_path, "collection.json")
+
+        if self._validate_stac_object(label_collection_json):
+
+            label_collection_item_hrefs = [
+                link["href"]
+                for link in self._read_json_data(label_collection_json)["links"]
+                if link["rel"] == "item"
+            ]
+            label_collection_item_hrefs = sorted(label_collection_item_hrefs)
+
+            for label_href in label_collection_item_hrefs:
+                label_json = os.path.join(label_collection_path, label_href)
+                indexed_item = self._load_items(label_json)
+                indexed_chips.append(indexed_item)
+
+        return indexed_chips
+
+    def _validate_bands(self, bands: List[str]) -> None:
+        """Validate list of bands.
+
+        Args:
+            bands: user-provided tuple of bands to load
+
+        Raises:
+            AssertionError: if ``bands`` is not a tuple
+            ValueError: if an invalid band name is provided
+        """
+        assert isinstance(bands, List), "The list of bands must be a tuple"
```

Comment:
mypy should handle type checking this. That said, does `bands` need to be a List, or can it be a more general `Sequence`?

Reply:
I have removed the assertion statement, and changed typing annotations for bands to `Sequence[str]`